packages/core/src/httpClient/retryableHttpClient.ts

```diff
@@ -15,14 +20,71 @@ export class RetryableHttpClient extends BasicHttpClient {
     super(baseUrl);
   }
 
+  public get<ResponseType>(
+    path: string,
+    options: RetryableHttpRequestOptions = {}
+  ) {
+    return super.get<ResponseType>(path, options);
+  }
+
+  public post<ResponseType>(
+    path: string,
+    options: RetryableHttpRequestOptions = {}
+  ) {
+    return super.post<ResponseType>(path, options);
+  }
+
+  public put<ResponseType>(
+    path: string,
+    options: RetryableHttpRequestOptions = {}
+  ) {
+    return super.put<ResponseType>(path, options);
+  }
+
+  public delete<ResponseType>(
+    path: string,
+    options: RetryableHttpRequestOptions = {}
+  ) {
+    return super.delete<ResponseType>(path, options);
+  }
+
+  public patch<ResponseType>(
+    path: string,
+    options: RetryableHttpRequestOptions = {}
+  ) {
+    return super.patch<ResponseType>(path, options);
+  }
+
+  protected async preRequest(
+    request: RetryableHttpRequest
+  ): Promise<RetryableHttpRequest> {
+    return super.preRequest(request);
+  }
+
+  protected async postRequest<T>(
+    response: HttpResponse<T>,
+    _: RetryableHttpRequest // eslint-disable-line
+  ): Promise<HttpResponse<T>> {
+    return super.postRequest<T>(response, _);
+  }
+
   protected async rawRequest<ResponseType>(
-    request: HttpRequest
+    request: RetryableHttpRequest
   ): Promise<HttpResponse<ResponseType>> {
     let retryCount = 0;
     // eslint-disable-next-line no-constant-condition
     while (true) {
       const response = await super.rawRequest<ResponseType>(request);
-      const shouldRetry = this.retryValidator(request, response, retryCount);
+
+      const retryValidator =
+        typeof request.retryValidator === 'function'
+          ? request.retryValidator
+          : this.retryValidator;
+
+      const shouldRetry =
+        retryCount < DEFAULT_MAX_RETRY_ATTEMPTS &&
```

Comment:
Do we want to make "make number of retries" a decision up to the validators?
We are passing `retryCount` to the retryValidators.
If yes, we don't need to export `DEFAULT_MAX_RETRY_ATTEMPTS`

Reply:
Ok, I hadn't seen @hhenriques's comment. Let's keep the max number of retries.